charts/revere/values.yaml

```diff
@@ -0,0 +1,211 @@
+# -- (object) Required; contents of revere.yaml to be given to the application
+config:
+
+# Global values used by this chart
+global:
+  # -- (string) What version of the application to deploy
+  applicationVersion: latest
+
+# name -- A name for the deployment that will be substituted into resource definitions
+name: revere
+
+imageConfig:
+  # -- (string) Image repository
+  repository: us-central1-docker.pkg.dev/dsp-artifact-registry/revere/revere
+  # -- (string) Image tag
+  # @default -- global.applicationVersion
+  tag: null
+  # -- (string) When to pull images
+  imagePullPolicy: Always
+
+# -- (number) Number of replicas for the deployment
+replicas: 3
+
+# -- (map) Annotations for application pods
+annotations: {}
+
+# -- (string) Optional override for the pod's runtimeClass
+runtimeClassName:
+
+resources:
+  requests:
+    # -- (string) Number of CPU units to request for the deployment
+    cpu: 500m
+    # -- (string) Memory to request for the deployment
+    memory: 4Gi
+  limits:
+    # -- (string) Number of CPU units to limit the deployment to
+    cpu: 500m
+    # -- (string) Memory to limit the deployment to
+    memory: 4Gi
+
+probes:
+  readiness:
+    # -- (boolean) If the readiness probe should be enabled
+    enabled: false
+    # -- (object) Spec for the readiness probe
+    spec:
+  liveness:
+    # -- (boolean) If the liveness probe should be enabled
+    enabled: false
+    # -- (object) Spec for the liveness probe
+    spec:
+  startup:
+    # -- (boolean) If the startup probe should be enabled
+    enabled: false
+    # -- (object) Spec for the startup probe
+    spec:
+
+secrets:
```

Comment:
I don't think you're supposed to put secrets in the value file. Normally you'd store these as environment variables or use helm secrets. You can read more about them [here](https://learning.oreilly.com/library/view/learn-helm/9781839214295/B15458_09_Final_JM_ePub.xhtml#_idParaDest-203)

Reply:
Of course! None of the fields here are meant to contain secrets themselves, they're just mechanisms for porting secrets into the deployment.

Workload Identity is a way for Google to allow Kubernetes service accounts to be linked to Google service accounts. Denis suggested we use it because it is more secure than storing an SA's private key in Vault. We might still use Vault for it, though, so that's why secrets-manager that we use everywhere else is also an option. I'm not calling it `vault` because I'm trying to be a bit forward-looking (other Vault mechanisms like the argocd-vault-plugin might require us to pass things differently).

In other words, the fields here are like `.Values.secrets.${The thing that is secret}.${How that thing gets into the deployment}`, so `.Values.secrets.gcpServiceAccount.workloadIdentity` configures Workload Identity to connect the GCP SA to the deployment. I just updated some of the comments here to make it a bit more clear.

I was admittedly having some trouble naming fields here, I'm open to any suggestions you have.